### ISSUE TITLE
remove syslog dependency

### ISSUE THREAD

**Issue by derekperkins** [vitessio/vitess#3563]
When not mounting the syslog as discussed with @enisoc, I see errors like these, but it appears that we're getting the information that needed to hit the syslog anyways, so I think we're ok.

```
cannot connect to syslog: Unix syslog delivery error
```

```
no connection, dropping syslog event: &events.TabletChange{Tablet:topodata.Tablet{Alias:(*topodata.TabletAlias)(0xc4201953c0), Hostname:"", PortMap:map[string]int32(nil), Keyspace:"urlslookup", Shard:"0", KeyRange:(*topodata.KeyRange)(nil), Type:2, DbNameOverride:"", Tags:map[string]string(nil), MysqlHostname:"", MysqlPort:3306}, Status:"updated"}
```

**Comment by enisoc** [comment]
@alainjobart What do you think about just deleting the syslogger plugin?

To be honest, the syslogger plugin was only added as an example of "what you could do with the event package". I don't think anyone really wants this info piped to syslog.

Internally we had a plugin that piped these events to a remote "global event logging" service. Ideally we should make an example plugin for something more analogous like fluentd.

**Comment by alainjobart** [comment]
I'd be fine removing it, it no one uses it.

For a sample one, we could just log into the regular logs with a prefix. And enable that by default, instead of syslog.

Well if we do that, we could keep the syslog one, just disabled by default...

**Comment by enisoc** [comment]
If you still want to make sure there's at least one example plugin, I'm file leaving it disabled by default.

**Comment by derekperkins** [comment]
I would want to export those logs by default in kubernetes for an audit trail

**Comment by aquarapid** [comment]
As running Vitess in k8s has been become more common, we are still logging this type of error for all events generated, since k8s pods almost never run syslog:

```
0730 23:10:58.958682       1 syslogger.go:84] no connection, dropping syslog event: &events........
```

Which just makes it hard to read.  Can we just convert all events to normal log.* messages if the syslogger cannot be contacted?  That way we only log a syslog-related error once (on startup), and that is it.

**Comment by aquarapid** [comment]
Should be addressed by #6511
